### PR TITLE
Implement and test the Home screen

### PR DESCRIPTION
**Pull request by ThomasMPont**
## Additions

- Implement the Home screen to match its most recent layout on Figma
- Test the Home screen's UI for 87% instruction coverage
- Add navigation graph entries for the new Home screen
- Set buttons to route from Home screen to Quest, Camera Feedback (MainAim), and Practice screens

## Changes

- Update navigation bar to route to Home screen
- Update login screen to route to Home screen after login

## Remarks

I have decided to group the Home screen's exercise selection buttons in a scrollable list, to account for future additions.

## Missing features

The camera preview functionality is not yet implemented.
The help button was not implemented, as it will be replaced by a common UI element soon.
The letter gesture dictionary list is not implemented, as a list of gestures does not yet exist.
The exercise buttons currently do not have any meaningful functionality.

## Screenshots

Jacoco test report.
image

Emulated Home screen.
image

### CODE REVIEW

File: app/src/main/java/com/github/se/signify/ui/screens/LoginScreen.kt
Comment (ziyadbagh) [inline app/src/main/java/com/github/se/signify/ui/screens/LoginScreen.kt:1]: As agreed the we route to the home screen after login in, good joob.

File: app/src/androidTest/java/com/github/se/signify/ui/screens/HomeScreenTest.kt
Comment (ziyadbagh) [inline app/src/androidTest/java/com/github/se/signify/ui/screens/HomeScreenTest.kt:1]: Well-organized test class, UI elements are displayed and clickable actions are well tested. It efficiently uses mock objects to verify navigation and interactions. Good job.

File: app/src/main/java/com/github/se/signify/ui/navigation/BottomNavigationMenu.kt
Comment (ziyadbagh) [inline app/src/main/java/com/github/se/signify/ui/navigation/BottomNavigationMenu.kt:1]: looks good to me

File: app/src/main/java/com/github/se/signify/ui/screens/HomeScreen.kt
Comment (ziyadbagh) [inline app/src/main/java/com/github/se/signify/ui/screens/HomeScreen.kt:1]: Clear structure for the HomeScreen composable, with clear modular components. Good job

**Review by SaFouNaldo — APPROVED** [review]
Everything works and the tests passes good job, but you need to format them.

**Review by nada-ezziti — APPROVED** [review]
**Pull request : Implement and test Home Screen** 

**Positive Aspects:**

1. Code clarity : The HomeScreen.kt file introduces well-structured composables, maintaining code clarity and scalability with future feature additions.

2. Testing : Test coverage ensures that all clickable elements perform their actions, and navigation flows are thoroughly tested.

3. No code duplication : It's great to see that the implementation avoids code duplication, which helps maintain clean, efficient, and easily manageable code. 

**Possible improvements :** 

1. Like you said we should replace somme buttons like the help one with shred composable later on.

2. We should implement the functionality of the streak counter.


______________________________________________

The Home screen is well-organized, with intuitive navigation to key sections like Quest, Camera Feedback, and Practice screens.  The layout is clean, and UI elements are easy to interact with, providing a smooth user experience. Overall, it's a solid implementation, and I believe it is ready to be merged.

**Pull request merged**